lib/grammers-client/src/types/media.rs

```diff
@@ -26,9 +33,48 @@ pub struct Uploaded {
 #[non_exhaustive]
 pub enum Media {
     Photo(Photo),
+    Document(Document),
     Uploaded(Uploaded),
 }
 
+impl Document {
+    pub(crate) fn from_media(
+        document: tl::types::MessageMediaDocument,
+        client: ClientHandle,
+    ) -> Self {
+        Self { document, client }
+    }
+
+    pub async fn download(&mut self, path: &Path) {
```

Comment:
1. This should return a `Result`, not panic on error.
2. We should do the same as `download_media_at_location` -- there's no reason not to.

```suggestion
    pub async fn download<P: AsRef<Path>>(&mut self, path: P) -> io::Result<()> {
```